pkg/controller/history/controller_history.go

```diff
@@ -49,10 +49,12 @@ import (
 const ControllerRevisionHashLabel = "controller.kubernetes.io/hash"
 
 // ControllerRevisionName returns the Name for a ControllerRevision in the form prefix-hash. If the length
-// of prefix is greater than 223 bytes, it is truncated to allow for a name that is no larger than 253 bytes.
+// of prefix is greater than 50 bytes, it is truncated to allow for a name that is no larger than 50 bytes.
+// This restriction is to ensure a standard Kubernetes name does not exceed the length limit of 63 bytes.
+// The controller revision hash is a Kubernetes label that must be 63 characters or less.
 func ControllerRevisionName(prefix string, hash string) string {
-	if len(prefix) > 223 {
-		prefix = prefix[:223]
+	if len(prefix) > 50 {
```

Comment:
why 50?

Reply:
Kubernetes label is restricted under 63 character https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set

The replica hash is an int32. Int32 is a 10  character. https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/history/controller_history.go#L92

There is a hyphen as a delimiter.

63 - 10 - 1 = 52 character. 
I will update to 52 character if this makes more sense.

Reply:
yes, we can put it at the limit.

But there are other open questions https://github.com/kubernetes/kubernetes/issues/64023#issuecomment-1521835790

Reply:
@alculquicondor I think we should use this PR to only address the current controller-revision-hash label limits for StatefulSet. The discussion to use annotation is a much bigger change. I replied to #64023 reasoning the label is used for per StatefulSet pod. Annotation is used for per replica set in Deployment.

Reply:
I don't think the implications of this PR are trivial.

Some people can be relying on the fact that the name is contained in the label value, when parsing.
If we merge this change as is, pod creations that previously failed will start succeeding. Is that the right solution? Not sure.

Reply:
looks like we can close this for now right?

/close